components/camel-azure/camel-azure-servicebus/src/main/java/org/apache/camel/component/azure/servicebus/ServiceBusConsumer.java

```diff
@@ -228,8 +232,21 @@ public void onFailure(Exchange exchange) {
             if (cause != null) {
                 getExceptionHandler().handleException("Error during processing exchange.", exchange, cause);
             }
+
             if (!getConfiguration().isDisableAutoComplete()) {
-                clientWrapper.abandon(message).subscribeOn(Schedulers.boundedElastic()).subscribe();
+                if (getConfiguration().isEnableDeadLettering()) {
```

Comment:
I think you need also to double check if there is actually a subqueue in your client instance.

Reply:
Yes your're right, thanks for the feedback.

Reply:
I've just removed my precedent comment as it was not correct.

According to the [Azure ServiceBus documentation](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-dead-letter-queues)
"Azure Service Bus queues and topic subscriptions provide a secondary subqueue, called a dead-letter queue"
So every queue or subscription has a deadletter.

but you are right, is the consumer is already consuming subqueue, we cannot deadletter the message. So I need to check if the subQueue option is set

Reply:
Yes, that was what I meant.

Reply:
I've added the check
`Objects.isNull(getConfiguration().getSubQueue()) || getConfiguration().getSubQueue().equals(SubQueue.NONE)`

Reply:
it's pushed